server/ERNI.PBA.Server.Host/Services/TransactionCalculator.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ERNI.PBA.Server.DataAccess.Model;
+using ERNI.PBA.Server.Host.Utils;
+
+namespace ERNI.PBA.Server.Host.Services
+{
+    public static class TransactionCalculator
+    {
+        public static IList<Transaction> Create(IEnumerable<TeamBudget> budgets, decimal distributedAmount)
+        {
+            var transactions = new List<Transaction>();
+            var availableBudgets = new Queue<TeamBudget>(budgets.OrderBy(x => x.Amount));
+            var amount = distributedAmount;
+            while (availableBudgets.Any())
+            {
+                var amountPerItem = amount / availableBudgets.Count;
+
+                var first = availableBudgets.Dequeue();
+                var amountToDeduct = Math.Min(amountPerItem, first.Amount).Round();
```

Comment:
Is it possible the round function will result in a higher number than the available budget?
Let's say the amountPerItem it 15, the first amount 12.9, then the result will be 13, right?